fluent-fallback/src/bundles.rs

```diff
@@ -0,0 +1,425 @@
+use crate::{
+    cache::{AsyncCache, Cache},
+    env::LocalesProvider,
+    errors::LocalizationError,
+    generator::{BundleGenerator, BundleIterator, BundleStream},
+    types::{L10nAttribute, L10nKey, L10nMessage},
+};
+use fluent_bundle::{FluentArgs, FluentBundle, FluentError};
+use std::borrow::Cow;
+
+pub enum BundlesInner<G>
+where
+    G: BundleGenerator,
+{
+    Iter(Cache<G::Iter, G::Resource>),
+    Stream(AsyncCache<G::Stream, G::Resource>),
+}
+
+pub struct Bundles<G>(BundlesInner<G>)
+where
+    G: BundleGenerator;
+
+impl<G> Bundles<G>
+where
+    G: BundleGenerator,
+    G::Iter: BundleIterator,
+{
+    pub fn prefetch_sync(&self) {
+        match &self.0 {
+            BundlesInner::Iter(iter) => iter.prefetch(),
+            BundlesInner::Stream(_) => todo!(),
+        }
+    }
+}
+
+impl<G> Bundles<G>
+where
+    G: BundleGenerator,
+    G::Stream: BundleStream,
+{
+    pub async fn prefetch_async(&self) {
+        match &self.0 {
+            BundlesInner::Iter(_) => todo!(),
```

Comment:
Same here, this could probably just prefetch sync and resolve I guess, but `todo!()` seems not too appropriate, shouldn't it just be `panic!("Can't prefetch a sync bundle set asynchronously")` or such?